onnxruntime/core/session/custom_ops.cc

```diff
@@ -279,6 +279,57 @@ ORT_API_STATUS_IMPL(OrtApis::KernelInfo_GetOutputTypeInfo, _In_ const OrtKernelI
   API_IMPL_END
 }
 
+ORT_API_STATUS_IMPL(OrtApis::KernelInfo_GetLogger, _In_ const OrtKernelInfo* info, _Outptr_ const OrtLogger** logger) {
```

Comment:
>const OrtKernelInfo* info, [](http://example.com/codeflow?start=56&length=26)

Getting the logger from KernelInfo may be helpful.

I was thinking of mimicking the way logging is done in our `standard` kernels. I think it is usually done via KernelContext that is passed to Compute().

And that one has a logger in it. [https://github.com/microsoft/onnxruntime/blob/main/include/onnxruntime/core/framework/op_kernel_context.h#L110](https://github.com/microsoft/onnxruntime/blob/main/include/onnxruntime/core/framework/op_kernel_context.h#L110)

So that way, the custom OP output would appear in the same logger as other kernels.

Reply:
The inference session sets the session logger on each execution provider. https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/session/inference_session.cc#L527

Separate searches with visual studio and grep indicate that this is the only place the EPs logger is set (for now).

Reply:
The logger set in the kernel context is different from the session logger set in the EP. See https://github.com/microsoft/onnxruntime/blob/e9ab56fa64c0644a2dc5287d1cd3b945bf7d7981/onnxruntime/core/session/inference_session.cc#L1985-L1987

Reply:
@yuslepukhin @pranavsharma 
Yes, this is correct. The kernel context does not use the session or session_state logger. It uses a "run logger" that may be different from the session logger.

I think the right move would be to add a similar KernelContext_GetLogger API so that we can get the appropriate logger during compute.